code/API_definitions/device-swap.yaml

```diff
@@ -0,0 +1,399 @@
+paths: 
+  /retrieve-date:
+    post:
+      tags:
+        - Retrieve Device Swap Date
+      description: Get timestamp of last device swap for a mobile user account provided with phone number.
+      operationId: retrieveDeviceSwapDate
+      security:
+        - openId:
+          - device-swap
+        - openId:
+          - device-swap:retrieve-date
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        description: |
+          Create a device swap date request for a phone number.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateDeviceSwapDate"
+        required: true
+      responses:
+        "200":
+          description: Contains information about Device swap change
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DeviceSwapInfo"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "429":
+          $ref: "#/components/responses/Generic429"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+        "504":
+          $ref: "#/components/responses/Generic504"
+  /check:
+    post:
+      tags:
+        - Check Device Swap
+      description: Check if device swap has been performed during a past period
+      operationId: checkDeviceSwap
+      security:
+        - openId:
+          - device-swap
+        - openId:
+          - device-swap:check
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        description: |
+          Create a check device swap request for a phone number.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateCheckDeviceSwap"
+        required: true
+      responses:
+        "200":
+          description: Returns whether a device swap has been performed during a past period
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/CheckDeviceSwapInfo"
+        "400":
+          $ref: "#/components/responses/Generic400Check"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "429":
```

Comment:
For consistency with other API and in particular with SIM swap I suggest to add error 422

```
    Generic422:
      description: Unprocessable Content
      headers:
        x-correlator:
          $ref: "#/components/headers/x-correlator"
      content:
        application/json:
          schema:
            $ref: "#/components/schemas/ErrorInfo"
          examples:
            GENERIC_422_NOT_SUPPORTED:
              description: Not Supported
              value:
                status: 422
                code: NOT_SUPPORTED
                message: Service not supported for this phoneNumber
            UNIDENTIFIABLE_PHONE_NUMBER:
              description: The phone number is not included in the request and the phone number information cannot be derived from the 3-legged access token
              value:
                status: 422
                code: UNIDENTIFIABLE_PHONE_NUMBER
                message: The phone number cannot be identified`

Reply:
included